agent.go

```diff
@@ -80,6 +80,10 @@ type container struct {
 	useSandboxPidNs bool
 }
 
+type sandboxStorage struct {
+	sourcePath string
```

Comment:
What is missing is a reference counter for each sandboxStorage. And since we want to use `sourcePath` to identify a sandbox storage, it's better to make it a map. Something like
```
type sandbox struct {
        <snip>
        storages map[string]sandboxStorage
}
```
And 
```
type sandboxStorage struct {
        refcount int
        [possibly other fields]
}
```
The container setup code path need to be modified such that if `sourcePath` is already mounted, do not mount it again. And the container cleanup code path also needs to be modified to make sure we do not unmount a `sourcePath` when there is still someone using it (IOW, refcount != 0).